BFAIR/io/database/mapping.py

```diff
@@ -0,0 +1,147 @@
+"""Mapping.
+
+Contains basic functions to read, merge, and write FIA MS mapping files.
+"""
+
+__all__ = ["from_struct", "merge", "read", "write"]
+
+import csv
+from collections import Iterable, namedtuple
+
+import pandas as pd
+
+from BFAIR.io.database.struct import FILE_COLUMNS as STRUCT_FILE_COLUMNS
+
+FILE_COLUMNS = ["unused_mass", "formula", "ids"]
```

Comment:
Thanks! 🎉

1. I will double check again, though I think all public methods had a test. However, I did not include tests for private methods/modules. For instance, there is a private `_queries` module that the rule library class depends on to create SQL statements. It is implicitly tested by the rule library tests (as most of the class' methods depend on SQL statements).
2. The logger is used by the rule library class to issue info/warning messages under certain conditions (e.g., no rules available after applying a stringent filter).
3. I somehow managed to fix this, so the tests have passed now.
4. I will ping Matthias so he can have a look. 😁

> Is there a reason for keeping these as individual functions instead of wrapping them into a class?

Since the struct/mapping files are handled as dataframes, I thought it made more sense to provide static methods (that return/take dataframes) instead of a container class (that wraps a dataframe).

Example usage:

```
from BFAIR.io import struct, mapping
# reaction rule simulations returning 'results' dict
offtarget_struct = struct.from_products_dict(results)
offtarget_mapping = mapping.from_struct(offtarget_struct)
expanded_mapping = mapping.merge(yeastGEM_mapping, offtarget_mapping)
```